kfp_notebook/pipeline/_notebook_op.py

```diff
@@ -228,6 +228,18 @@ def __init__(self,
             gpu_vendor = self.pipeline_envs.get('GPU_VENDOR', 'nvidia')
             self.container.set_gpu_limit(gpu=str(gpu_limit), vendor=gpu_vendor)
 
+        # Attach metadata to the pod
+        # Node type (a static type for this op)
+        self.add_pod_label('elyra-node-type', 'notebook-script')
+        # Pipeline name
+        # Should we use kwargs.get('pipeline_name')?
+        self.add_pod_label('elyra-pipeline-name',
+                           self.cos_directory)
```

Comment:
1. The pipeline _experiment_ name is passed  as parameter `cos_directory`. If for whatever reason the value no longer represents the _experiment_ name, the attached metadata is invalid. For example, let's say in the future the `cos_directory` value is a UUID then this breaks.
2. I would suggest to introduce a required parameter `experiment_name` and use that one instead in this place. This does require changes in the `kfp_processor`.

Reply:
See https://github.com/elyra-ai/elyra/blob/master/elyra/pipeline/processor_kfp.py#L53 why `cos_directory` does not represent the pipeline name that the user entered.

Reply:
I agree.  We need to stop overloading items here and this should be revisited.

Should the experiment name be the timestamp-suffixed value of pipeline name or just pipeline name?  I kinda think it should represent an _experiment_ or _run_ of a pipeline and therefore the former and then add a label for `elyra-experiment` so that all pods relative to a given run could be queried, as well as all pods for a given pipeline (that spans experiments).

Reply:
Are you saying we should expose both - the pipeline name and the experiment name? While they are currently identical, that might (and hopefully will) change in the future when we take advantage of pipeline versioning.

Reply:
I view the pipeline name to be what the user entered and the experiment name to be the timestamp-suffixed value that is essentially _system-owned_ and changes on each submission/run - both of which are meaningful from a labeling perspective.

> when we take advantage of pipeline versioning

I would hope we'd introduce a new property to reflect a pipeline's version.

Reply:
Should be resolved. Introduced `pipeline_version` as an optional input parameter, which defaults to the empty string